modules/andes-core/broker/src/main/java/org/wso2/andes/server/handler/QueueDeleteHandler.java

```diff
@@ -85,60 +86,53 @@ public void methodReceived(AMQStateManager stateManager, QueueDeleteBody body, i
+                    AMQSessionModel session = queue.getExclusiveOwningSession();
+                    if (queue.isExclusive() && !queue.isDurable()
+                        && (null == session || session.getConnectionModel() != protocolConnection)) {
+                        throw body.getConnectionException(AMQConstant.NOT_ALLOWED,
+                                                          "Queue " + queue.getNameShortString()
+                                                          + " is exclusive, but not created on this Connection.");
+                    }
 
-                boolean isQueueDeletable = ClusterResourceHolder.getInstance().
-                        getVirtualHostConfigSynchronizer().checkIfQueueDeletable(queue);
+                    boolean isQueueDeletable = ClusterResourceHolder.getInstance().
+                            getVirtualHostConfigSynchronizer().checkIfQueueDeletable(queue);
 
-                if(isQueueDeletable)
-                {
-                    int purged = queue.delete();
+                    if (isQueueDeletable) {
+                        int purged = queue.delete();
 
-                    if (queue.isDurable())
-                    {
-                        store.removeQueue(queue);
+                        if (queue.isDurable()) {
+                            store.removeQueue(queue);
 
-                        //tell Andes Kernel to remove queue
-                        QpidAndesBridge.deleteQueue(queue);
+                            //tell Andes Kernel to remove queue
+                            QpidAndesBridge.deleteQueue(queue);
+                        }
 
+                        MethodRegistry methodRegistry = protocolConnection.getMethodRegistry();
+                        QueueDeleteOkBody responseBody = methodRegistry.createQueueDeleteOkBody(purged);
+                        protocolConnection.writeFrame(responseBody.generateFrame(channelId));
+                    } else {
+                        _logger.warn("Cannot Delete Queue" + queue.getName() + " It Has Registered Subscriptions.");
+                        throw new AMQException(
+                                "Cannot Delete Queue" + queue.getName() + " It Has Registered Subscriptions.");
                     }
-
-                    MethodRegistry methodRegistry = protocolConnection.getMethodRegistry();
-                    QueueDeleteOkBody responseBody = methodRegistry.createQueueDeleteOkBody(purged);
-                    protocolConnection.writeFrame(responseBody.generateFrame(channelId));
-                }
-                else
-                {
-                    _logger.warn("Cannot Delete Queue" + queue.getName() + " It Has Registered Subscriptions.");
-                    throw new AMQException("Cannot Delete Queue" + queue.getName() + " It Has Registered Subscriptions.");
                 }
             }
+        } else {
+            _logger.warn("Deletion of Dead letter channel not permitted.");
```

Comment:
From the server's side, is it an error? I thought it should be a warning saying somebody attempted to delete the DLC and it is not permitted.